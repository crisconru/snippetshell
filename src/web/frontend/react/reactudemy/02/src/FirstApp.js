import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

const FirstApp = ( { coletilla, subtitulo } ) => {
    const saludo = 'Hola mundo'
    return (
        <Fragment>
            <h1>{ `${saludo} ${coletilla}` }</h1>
            <p>{ subtitulo }</p>
        </Fragment>
    )
}

FirstApp.propTypes = {
    coletilla: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default FirstApp 