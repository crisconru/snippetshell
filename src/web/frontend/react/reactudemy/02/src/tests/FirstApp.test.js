import React from 'react';
//import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import FirstApp from '../FirstApp'


describe('Pruebas FirstApp', () => {
    /*
    test('FirstApp debe devolver "Hola mundo zanguango"', () => {
        const resultado = 'Hola mundo zanguango'
        const col = 'zanguango'
        const { getByText } = render( <FirstApp coletilla={col}/> )
        expect( getByText(resultado) ).toBeInTheDocument()
    })
    */
    test('Debe mostrar <FirstApp/> con coletilla', () => {
        const col = 'zanguango'
        const componente = shallow( <FirstApp coletilla={col}/> )
        expect(componente).toMatchSnapshot()
     })
     test('Debe mostrar <FirstApp/> con subtitulo', () => {
        const col = 'zanguango'
        const sub = 'mangango'
        const componente = shallow( <FirstApp coletilla={col} subtitulo={sub}/> )
        const parrafo = componente.find('p').text()
        //console.log(parrafo)
        expect(parrafo).toBe(sub)
     })
})