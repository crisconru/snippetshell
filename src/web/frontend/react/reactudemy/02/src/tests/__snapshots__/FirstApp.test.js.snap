// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas FirstApp Debe mostrar <FirstApp/> con coletilla 1`] = `
<Fragment>
  <h1>
    Hola mundo zanguango
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
